src/Microsoft.Azure.WebJobs.Host.Storage/Singleton/BlobLeaseDistributedLockManager.cs

```diff
@@ -93,61 +89,59 @@ public async virtual Task<string> GetLockOwnerAsync(string account, string lockI
                 await WriteLeaseBlobMetadata(lockBlob, leaseId, lockOwnerId, cancellationToken);
             }
 
-            SingletonLockHandle lockHandle = new SingletonLockHandle(leaseId, lockId, lockBlob, lockPeriod);
+            SingletonLockHandle lockHandle = new SingletonLockHandle(leaseId, lockId, this.GetBlobLeaseClient(lockBlob, leaseId), lockPeriod);
 
             return lockHandle;
         }
 
-        protected abstract CloudBlobContainer GetContainer(string accountName);
-
-        internal CloudBlobDirectory GetLockDirectory(string accountName)
+        protected virtual BlobContainerClient GetContainerClient(string accountName)
         {
-            if (string.IsNullOrEmpty(accountName))
+            if (!string.IsNullOrWhiteSpace(accountName))
             {
-                accountName = string.Empty; // must be non-null for a dictionary key
+                throw new InvalidOperationException("Must replace singleton lease manager to support multiple accounts");
```

Comment:
Probably should change this to "Must replace IDistributedLockManager...". Also, I can't remember the context for this error - we'll need to review to make sure things still work as intended in this regard